tss-esapi/src/abstraction/transient.rs

```diff
@@ -645,9 +686,9 @@ impl TransientKeyContextBuilder {
             .create_primary(
                 self.hierarchy,
                 &create_restricted_decryption_rsa_public(
-                    self.default_context_cipher.try_into()?,
+                    self.default_context_cipher,
                     root_key_rsa_key_bits,
-                    RsaExponent::default(),
+                    RsaExponent::create(0).unwrap(),
```

Comment:
That's a bit unfortunate change. I guess this is related to the fact that RsaExponent's Default implementation now uses `value: (1 << 16) + 1` and not `0`?

[The spec](https://trustedcomputinggroup.org/wp-content/uploads/TCG_TPM2_r1p59_Part2_Structures_pub.pdf) in paragraph 12.2.3.5 (TPMS_RSA_PARMS) says that:

> An exponent of zero indicates that the exponent is the default of 2^16 + 1.

It's just... changing the `default()` to `RsaExponent::create(0).unwrap()` looks like a small code smell... maybe that's just me :)

Reply:
Yes, similar to Paul's comment above about `PublicKeyRsa::new_empty_with_size(rsa_key_bits)` => `PublicKeyRsa::default()`, just in the opposite direction. If we use the new one the attributes of the key aren't kept the same and the primary key names becomes something else.

Reply:
:thinking: I think in this case 0 is exactly the same as 2^16 + 1 (== `(1 << 16) + 1`) by the TPM spec. It's just `create(0).unwrap()` triggers a red-flag in my mind as 0 is defined by the spec to be a good default and unwraps... are just suspicious :) But this is just a minor thing. I don't want to split hair on something this small :)

Reply:
It's not exactly the same only because 0 is not the same number as 2^16+1, even if they're treated the same by the TPM. The reason it's a problem is that when the cryptographic hash is computed over the `TPM2B_Public` structure the fact that it's some number instead of 0 completely changes the result. 

Yeah, we tend to avoid `unwrap`, but it's accepted in cases where we know it isn't going to fail. Tbh we could have a test somewhere making sure that `RsaExponent::create(0).unwrap` doesn't panic, since 0 is a fairly standard value. I'm debating changing the `RsaExponent::default` to 0, though, thoughts? @Superhepper

Reply:
If the default value used by RsaExponent is wrong why not change it? Or add ``` const ZERO_EXPONENT: RsaExponent = RsaExponent { value: 0}``` to the RsaExponent struct.

Reply:
It's not really wrong per se, `(1<<16) + 1` is equally correct. I'll add the `const`, was mulling that idea over.

Reply:
The architecture spec does state that the default should be 2^16+1 in a few places, so I'll leave it as is, I think 0 is just the lazy/shorthand way of doing the same thing.